### ISSUE TITLE
loadData - NullPointerException when type not provided and column name invalid

### ISSUE THREAD

**Issue by kdebski85** [liquibase/liquibase#1687]
## Environment
**Liquibase Version**: 4.2.2

**Liquibase Integration & Version**: maven

**Liquibase Extension(s) & Version**: 

**Database Vendor & Version**: H2

**Operating System Type & Version**: Windows 10

## Description
When loadData has a column without type defined, it tries to determine type from DB metadata (method `retrieveMissingColumnLoadTypes`).
When the column name is invalid, NullPointerException is thrown.

It is problematic when loadData has many columns. It takes time to find invalid column.

## Steps To Reproduce
Add loadData change with column without type provided and with invalid column name:

```
        <loadData  file="foo.csv" relativeToChangelogFile="true" separator="," tableName="FOO">
            <column header="INVALID" name="INVALID"/> <!-- not type provided, there is no column INVALID in FOO table -->
        </loadData>
```

## Actual Behavior
When loadData has a column without type defined, it tries to determine type from DB metadata.
When the column name is invalid, NullPointerException is thrown.

liquibase.change.core.LoadDataChange code (lines from 659):

```
            DataType dataType = tableColumns.get(entry.getKey()).getType(); //tableColumns.get(entry.getKey()) returns null and getType() causes NPE
            if (dataType == null) {
                LOG.warning(LogType.LOG, String.format(coreBundle.getString("unable.to.find.load.data.type"),
                    columnConfig.toString(), snapshotOfTable.toString()));
                columnConfig.setType(LOAD_DATA_TYPE.STRING.toString());
            }
```

## Expected/Desired Behavior
loadData should throw an exception with invalid column name instead of NullPointerException or log warning

```
            Column column= tableColumns.get(entry.getKey());
            if (column == null || column.getType() == null) {
                LOG.warning(LogType.LOG, String.format(coreBundle.getString("unable.to.find.load.data.type"),
                    columnConfig.toString(), snapshotOfTable.toString()));
                columnConfig.setType(LOAD_DATA_TYPE.STRING.toString());
            }
```

**Comment by molivasdat** [comment]
Thanks @kdebski85 for writing up this enhancement request.  We will add it to the list of issues that we are processing.

**Comment by wertklop** [comment]
Fixed by 4.4.5 version, is not it?

**Comment by FBurguer** [comment]
Issue is already fixed for liquibase `4.15`. Try updating to our latest version. If the issue continues for you, please re-open the issue and we will look more into it. Thanks!
